### ISSUE TITLE
"WAIT ON PATH" lock

### ISSUE THREAD

**Issue by anatol** [bfleischer/macfuse#4]
Hi,

Here is a follow up for the issue #3. Testing my app with your master branch and there is yet another issue:

unique: 2, opcode: ACCESS (34), nodeid: 409, insize: 48
access /Users/anatol/sources/tup/@tupjob-28/libtup_client.a 02
   unique: 2, success, outsize: 16
unique: 1, opcode: ACCESS (34), nodeid: 409, insize: 48
access /Users/anatol/sources/tup/@tupjob-28/libtup_client.a 00
   unique: 1, success, outsize: 16
unique: 3, opcode: UNLINK (10), nodeid: 408, insize: 56
QUEUE PATH 409 (w)
WAIT ON PATH 409 (w)
/usr/bin/ranlib: can't create output file: libtup_client.a (Bad file descriptor)
ar: internal ranlib command failed

The "WAIT ON PATH" message comes from http://www.google.com/codesearch/p?hl=en#rdCCvgMofJA/trunk/rdp/fuse/fuse-2.8.4/lib/fuse.c&q=%22WAIT%20ON%20PATH%22&sa=N&cd=1&ct=rc&l=645

Here is the code that originally added the logic https://github.com/anatol/fuse/commit/72faaa58a85a0c8de724e8c22a4ec5b9f75e42e3

I am going to recompile 2.1.1 and see if it is reproducible on that version.

**Comment by anatol** [comment]
No, I do not see the problem on 2.1.11 branch, so I guess it is related somehow to the fuse-2.8.5 changes.

**Comment by anatol** [comment]
Here is the log from macfuse 2.2.0 https://gist.github.com/969715

**Comment by bfleischer** [comment]
Fixed! 

libfuse 2.8.5 uses a more advanced locking mechanism compared to libfuse 2.7.3. When "applying" the libfuse 2.7.3 patch to libfuse 2.8.5 I kept using the old locking mechanism for MacFUSE specific functions (setattr_x, setvolname, exchange and getxtimes). This lead to the deadlocks.

Finally the tup bootstrap script completes without an error.

**Comment by anatol** [comment]
You are totally rock, man!
